master/buildbot/status/github.py

```diff
@@ -14,52 +14,96 @@
 # Copyright Buildbot Team Members
 from __future__ import absolute_import
 
-import datetime
+from datetime import datetime
 
 from twisted.internet import defer
 from twisted.python import log
-from txgithub.api import GithubApi as GitHubAPI
+try:
+    from txgithub.api import GithubApi as GitHubAPI
+except ImportError:
+    GitHubAPI = None
 from zope.interface import implements
 
+from buildbot import config
 from buildbot.interfaces import IStatusReceiver
 from buildbot.process.properties import Interpolate
 from buildbot.status.base import StatusReceiverMultiService
 from buildbot.status.builder import FAILURE
 from buildbot.status.builder import SUCCESS
 
+_STATE_MAP = {
+    SUCCESS: 'success',
+    FAILURE: 'failure',
+}
 
-class GitHubStatus(StatusReceiverMultiService):
-    implements(IStatusReceiver)
 
+def _getGitHubState(results):
+    """
+    Convert Buildbot states into GitHub states.
+    """
+    # GitHub defines `success`, `failure` and `error` states.
+    # We explicitly map success and failure. Any other BuildBot status
+    # is converted to `error`.
+    return _STATE_MAP.get(results, 'error')
+
+
+def _timeDeltaToHumanReadable(start, end):
```

Comment:
There is nothing specific to Github here.
That could be used by other status plugins.